src/core_plugins/kibana/public/management/sections/indices/create_index_pattern/create_index_pattern.js

```diff
@@ -184,14 +203,15 @@ uiModules.get('apps/management')
 
   this.createIndexPattern = () => {
     const id = this.newIndexPattern.name;
-    const timeFieldName =
-      this.newIndexPattern.isTimeBased
-      ? this.newIndexPattern.timeField.name
-      : undefined;
+    let timeFieldName;
+    if ((this.newIndexPattern.timeField !== TIME_FILTER_FIELD_OPTIONS.NO_DATE_FIELD_SELECTED)
+      && (this.newIndexPattern.timeField !== TIME_FILTER_FIELD_OPTIONS.NO_DATE_FIELDS_IN_INDEX)) {
```

Comment:
I think this could be reduced to:

```javascript
![
  TIME_FILTER_FIELD_OPTIONS.NO_DATE_FIELD_SELECTED,
  TIME_FILTER_FIELD_OPTIONS.NO_DATE_FIELDS_IN_INDEX
].includes(this.newIndexPattern.timeField)
```